### PR TITLE
Remove uses of `Data.Nat.Solver` from a number of places

### PR DESCRIPTION
**Pull request by JacquesCarette**
Various places gratuitously used `Data.Nat.Solver` when some fairly simple direct proofs could be used instead. This made `Data.Rational` (amongst others) very 'high' on the dependency graph when this was not needed.

### CODE REVIEW

File: src/Data/Integer/Properties.agda
Comment (MatthewDaggitt) [inline src/Data/Integer/Properties.agda:1393]: I'm not sure whether having half the cases in one file and the other set of equally long cases in this file, is ideal? I think it would be better (everywhere!) just to expand out the cases locally?

I know GCD.Lemmas doesn't do so, but that is a very old file?

File: src/Data/Integer/Properties.agda
Comment (MatthewDaggitt) [inline src/Data/Integer/Properties.agda:1354]: Likewise here better to have a local lemma?

**Review by MatthewDaggitt — COMMENTED** [review]
> Various places gratuitously used Data.Nat.Solver when some fairly simple direct proofs could be used instead.

Having looked at the changes, I'm not sure if some of these can be called "simple". I wonder if a parallel effort might be to reduce the dependency foot print of the solvers instead? Fundamentally the solvers should use be able to use the boolean decision procedures in the `Base` modules rather than `Dec` decision procedures in `Properties` right?

**Comment by Taneb** [comment]
> > Various places gratuitously used Data.Nat.Solver when some fairly simple direct proofs could be used instead.
> 
> Having looked at the changes, I'm not sure if some of these can be called "simple". I wonder if a parallel effort might be to reduce the dependency foot print of the solvers instead? Fundamentally the solvers should use be able to use the boolean decision procedures in the `Base` modules rather than `Dec` decision procedures in `Properties` right?

You'd still need all the algebra laws, which still brings in `.Properties`...

**Comment by MatthewDaggitt** [comment]
> You'd still need all the algebra laws, which still brings in .Properties...

Hmm good point!

**Comment by JacquesCarette** [comment]
> Having looked at the changes, I'm not sure if some of these can be called "simple". 

Compared to a lot of proofs in `agda-categories`, they are... but point taken. Though I'd say that many of them are no worse than some of the monsters in the same files. (`Data.Integer.Properties` is seriously frightening; I'll open some issues later.)

File: src/Data/Rational/Unnormalised/Properties.agda
Comment (JacquesCarette) [reply src/Data/Rational/Unnormalised/Properties.agda:1126]: Will indeed get rid of them when I spot them.

File: src/Data/Integer/Properties.agda
Comment (JacquesCarette) [reply src/Data/Integer/Properties.agda:1393]: I'm fine with pulling these in.

I do think if we had a big `Lemmas` file, we'd notice quite a few patterns, and could shrink things. But that can be a v3.0 kind of task.

Comment (JacquesCarette) [reply src/Data/Integer/Properties.agda:1393]: And I now remember why I put these in a separate file: inside `Integer.Properties`, `+` and `*` are for integers, and the ones for naturals are 'long'. So these proofs end up looking much, much worse. But inside a different context, i.e. such as a separate module, this is not a problem.

So, in looking again, I'd rather not do this.

File: src/Data/Digit.agda
Comment (gallais) [inline src/Data/Digit.agda:103]: The alignment of the LHSs got messed up here.

File: src/Data/Fin/Properties.agda
Comment (gallais) [inline src/Data/Fin/Properties.agda:675]: Alignment issues here too (for the reasoning combinators this time).

File: src/Data/Digit.agda
Comment (JacquesCarette) [reply src/Data/Digit.agda:103]: Actually, this is intentional -- this is a nested proof, see the 'begin' on the line above.

File: src/Data/Fin/Properties.agda
Comment (JacquesCarette) [reply src/Data/Fin/Properties.agda:675]: That's odd, I don't see any alignment problems in my emacs!

File: src/Data/Integer/Properties.agda
Comment (MatthewDaggitt) [reply src/Data/Integer/Properties.agda:1393]: Okay, I withdraw my objections.

**Review by MatthewDaggitt — APPROVED** [review]
Once the merge conflict is resolved happy to merge.

**Comment by JacquesCarette** [comment]
(Normally don't merge my own, but I take the above as permission to do so rather than waiting.)

**Pull request merged**
